cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -452,6 +452,7 @@ function(ADD_TEST_CASE REL_TEST_NAME)
     add_test(${TEST_NAME}
       bash -c "cd '${CMAKE_SOURCE_DIR}'; \
                valgrind --suppressions=valgrind.supp --tool=memcheck --gen-suppressions=all \
+                 --track-origins=yes \
```

Comment:
According to the Valgrind documentation:
> Performance overhead: origin tracking is expensive. It halves Memcheck's speed and increases memory use by a minimum of 100MB, and possibly more. Nevertheless it can drastically reduce the effort required to identify the root cause of uninitialised value errors, and so is often a programmer productivity win, despite running more slowly.

I'm not sure we want to make CI builds slower than they need to be. Is this necessary in your opinion?

Reply:
I'd say only enable this in local, if this doesn't increase the coverage of finding new bugs.

Reply:
Yes. I'll remove this

Reply:
This should be fine. It can be a local env addition, if valgrind reports some errors.